### ISSUE TITLE
Discovering tests in watch task

### ISSUE THREAD

**Issue by steveoh** [formulahendry/vscode-dotnet-test-explorer#124]
I have `dotnet watch` setting enabled so that tests are run on save. I am noticing that if I add a new test **file** it does not get picked up until I restart the watch task by reloading vs code. Is this a test explorer issue or expected behavior with how dotnet watch works?

**Comment by stefanforsberg** [comment]
Just to clarify, by "not get picked up" you mean that it's not shown in the test tree right?

As it's currently working the extension only generates the tree upon test discovery (which happens when you start the extensions or refresh it via the icon in the top menu). The results from the test run only updates the icon (pass, fail etc).

Ideally we'd update the tree with new tests if a test run results in **new** tests the tree doesn't know about but it requires some thought and time. We also can get into interesting edge cases, say for instance you change a namespace and the tree shows both the old and the new namespace etc.

tldr; I want to make it better but I'm not sure if or how or when =)

**Comment by steveoh** [comment]
Hmm, I'll have to look at the tree. I was just noticing that the command that is run isn't scanning for the new test file. So that must be a dotnet cli issue. It's probably a similar issue to what you are facing.

If I start dotnet watch it will pick up all tests in all files. If I add tests to an existing file, those will be discovered and executed. But when I add a new cs file with tests they are not discovered or executed. 

The file is compiled so i would think it would be the same scenario as adding tests to an existing file. There must be some caching or something going on.

**Comment by stefanforsberg** [comment]
Hmm yea, I just gave it a quick try here and when I add a new file it does not trigger a watch reaction before I restart the dotnet watch command, did not know that =)

If it is a limitation with the cli tool there's not much we can do. What we could do is to restart the dotnet watch process when refreshing the tree though.

**Comment by steveoh** [comment]
I think that is an acceptable approach. Saves me from having to reload the entire window.

**Comment by stefanforsberg** [comment]
@steveoh 0.5.2 was just released and it handles new tests better (eg update the tree with tests that we're added after the extensions was started). It still does not restart the dotnet watch process, but I think the workaround mentioned in the issue you linked (touching another file that is watch and saving it) is a good enough workaround. Hopefully it is eventually fixed in the cli.
